app/tests/unit/components/validation/middleware.spec.js

```diff
@@ -0,0 +1,64 @@
+
+const bytes = require('bytes');
+const config = require('config');
+
+const { middleware, modelValidation } = require('../../../../src/components/validation');
+
+const maxFileSize = bytes.parse(config.get('carbone.uploadSize'));
+
+describe('validateCarbone', () => {
+  const carboneSpy = jest.spyOn(modelValidation, 'carbone');
+  const handleValidationErrorsSpy = jest.spyOn(middleware, '_handleValidationErrors');
+
+  beforeEach(() => {
+    carboneSpy.mockReset();
+    handleValidationErrorsSpy.mockReset();
+  });
+
+  afterAll(() => {
+    carboneSpy.mockRestore();
+    handleValidationErrorsSpy.mockRestore();
+  });
+
+  it('should call modelValidation.carbone and _handleValidationErrors', () => {
+    const req = { body: {} };
+    const fn = () => {};
+    carboneSpy.mockReturnValue([]);
+
+    middleware.validateCarbone(req, {}, fn);
+
+    expect(carboneSpy).toHaveBeenCalledTimes(1);
+    expect(carboneSpy).toHaveBeenCalledWith(req.body);
+    expect(handleValidationErrorsSpy).toHaveBeenCalledTimes(1);
+    expect(handleValidationErrorsSpy).toHaveBeenCalledWith({}, fn, []);
+  });
+});
+
+describe('validateTemplate', () => {
+  const templateSpy = jest.spyOn(modelValidation, 'template');
+  const handleValidationErrorsSpy = jest.spyOn(middleware, '_handleValidationErrors');
+
+  beforeEach(() => {
+    templateSpy.mockReset();
+    handleValidationErrorsSpy.mockReset();
+  });
+
+  afterAll(() => {
+    templateSpy.mockRestore();
+    handleValidationErrorsSpy.mockRestore();
+  });
+
+  it('should call modelValidation.template and _handleValidationErrors', () => {
+    const req = { body: {} };
+    const fn = () => {};
+    templateSpy.mockReturnValue([]);
+
+    middleware.validateTemplate(req, {}, fn);
+
+    expect(templateSpy).toHaveBeenCalledTimes(1);
+    expect(templateSpy).toHaveBeenCalledWith(req.body, maxFileSize);
+    // TODO: Figure out why this test spy isn't working
+    // expect(handleValidationErrorsSpy).toHaveBeenCalledTimes(1);
```

Comment:
Should we remove this since the test above is working?

Reply:
I left in this breadcrumb because we ideally want to be testing to make sure that the other handleValidationErrors function is actually being invoked. I wasn't able to get this to work in the unit test - likely due to some strange scoping issue, but would be good to have this revisited in the future.